rocprim/include/rocprim/config.hpp

```diff
@@ -77,28 +77,42 @@
 #undef ROCPRIM_TARGET_CDNA1
 #undef ROCPRIM_TARGET_CDNA2
 #undef ROCPRIM_TARGET_CDNA3
+#undef ROCPRIM_TARGET_UNKNOWN
 
 // See https://llvm.org/docs/AMDGPUUsage.html#instructions
-#if defined(__gfx940__) || defined(__gfx941__) || defined(__gfx942__)
+#if defined(__gfx940__) || defined(__gfx941__) || defined(__gfx942__) || defined(__gfx9_4_generic__)
     #define ROCPRIM_TARGET_CDNA3 1
 #elif defined(__gfx90a__)
     #define ROCPRIM_TARGET_CDNA2 1
 #elif defined(__gfx908__)
     #define ROCPRIM_TARGET_CDNA1 1
 #elif defined(__gfx900__) || defined(__gfx902__) || defined(__gfx904__) || defined(__gfx906__) \
-    || defined(__gfx90c__)
+    || defined(__gfx90c__) || defined(__gfx9_generic__)
     #define ROCPRIM_TARGET_GCN5 1
-#elif defined(__GFX12__)
+#elif defined(__GFX12__) || defined(__gfx12_generic__)
     #define ROCPRIM_TARGET_RDNA4 1
-#elif defined(__GFX11__)
+#elif defined(__GFX11__) || defined(__gfx11_generic__)
```

Comment:
It does no harm to do this, but I will note that the `__GFX11__` macro is defined whenever `__gfx11_generic__` is defined.

Reply:
https://cuda.godbolt.org/z/fY54o1ohT